### ISSUE TITLE
Anki not respecting time windows

### ISSUE THREAD

**Issue by Mornon** [ankidroid/Anki-Android#5449]
I have a deck with, for new cards, 10 20 30 60 120 900 240 1440 steps; the problem is, while Anki does visualize the right time on the buttons, it doesn't really respect them: when this morning I hit the "20 min." buttons, Anki showed me the card hours later. Then again I hit "30 min.", I had to wait two hours and a half for it to show again (and it showed together with a review, not sure if it matters).

On another deck I have 10 20 40 1440; after not remembering a card I hit "Again - 10 min.", and it showed again after half a hour. I then hit "20 min.", and that was respected.

The issue seems limited to new cards, since I checked a couple of cards already out of study status and (at least from Anki Desktop browser) the due date was updated to the right day.

The problem is, I have no idea why this is happening, also because it seems to act differently according to the deck (just on "Again" for one deck, also with "Good" for the other), and when I tried with a brand-new deck the issue doesn't seem to appear, so I can't really give a step-by-step process to reproduce it; I tried to search for similar issues, but I didn't find any, so I was hoping maybe someone has any idea what could be happening, or how I can analyze it to give more information.

Edit: I just thought I could maybe try to make new decks with the same settings and then just move all the cards there to see if the issue persists; I'll try that and see what happens, but I'm leaving this here to see if someone has ideas that maybe can solve it in a easier way, and just in case it could help in identifying a bug.

###### Debug info
AnkiDroid Version = 2.8.4

Android Version = 6.0.1

ACRA UUID = 98e79a32-2044-4bbb-a788-870a4efe0494

###### Research
*Enter an [ x ] character to confirm the points below:*

[ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[ x ] I have searched for similar existing issues here and on the user forum

**Comment by mikehardy** [comment]
Hi there! I wouldn't pursue any in-depth troubleshooting on the 2.8 release series anymore, 2.9 is very close to release and should be used as the basis for any detailed testing like this as it has seen some scheduler activity: https://docs.ankidroid.org/manual.html#betaTesting

**Comment by Mornon** [comment]
Thanks, I'll wait for 2.9 then; is there any timeline on it?

**Comment by mikehardy** [comment]
You can subscribe here #4961 but from our perspective I'm not sure any remaining code work will actually be done, I think at this point it's mostly documentation then we'll make it the full release, with anything else deferred to 2.10

From my perspective It's ready to go on the code side - not perfect across all 1+ million users but fixes lots of bugs people experience on 2.8 and seems more stable in general because of that. Has about 2,000 users in the beta channel already, FWIW

**Comment by Mornon** [comment]
This is still happening with 2.9: I hit "20 min." a hour and a half ago, still waiting. I'll try to make new decks with the same settings and then just move all the cards there to see if the issue persists, I'll keep you updated.

**Comment by Mornon** [comment]
The issue is still there: "20 min." button, four and a half hour of waiting. I tried creating a new option group and making new decks in which I moved the cards. I have no idea why it's happening.

**Comment by Mornon** [comment]
I tried a bunch of things and I think I understood something, not sure if it's just that (I'll say the possible issue, then everything I tried for reference).

What I found is that if I reset a card and that card is in a subdeck, then every new card in that subdeck will have issue with time steps; still it's not all, since then again I did a new deck, moved there all my cards, created a new card, and it has the same issue.

What I tried: I tried using the steps I have on another deck that doesn't present this issue, then I tried using "10 5", but I gave up after 6 hours (instead of 5 minutes) and just deleted the card.

I tried to create the same card in a new deck in the same option group and it worked, then I created in again in the deck with the issue, to no avail: 5 minutes expected, after 10 minutes nothing.

So it seems new decks aren't affected, yet again I tried creating a new deck and moving there all my cards, and the issue still holds. Then I tried yet another new deck, this time with "10 1", and it didn't work. I was wondering if the issue could derive from one step being lower the the previous one, but since "10 5" sometimes work it doesn't seem to be that.

This is something that also affectes Anki Desktop; if something else come to mind I'll update here, but I really have no clue about what's happening and I have no idea about how to further investigate this.

**Comment by Mornon** [comment]
I tried a bunch of things and I think I understood something, not sure if it's just that (I'll say the possible issue, then everything I tried for reference).

What I found is that if I reset a card and that card is in a subdeck, then every new card in that subdeck will have issue with time steps; still it's not all, since then again I did a new deck, moved there all my cards, created a new card, and it has the same issue.

What I tried: I tried using the steps I have on another deck that doesn't present this issue, then I tried using "10 5", but I gave up after 6 hours (instead of 5 minutes) and just deleted the card.

I tried to create the same card in a new deck in the same option group and it worked, then I created in again in the deck with the issue, to no avail: 5 minutes expected, after 10 minutes nothing.

So it seems new decks aren't affected, yet again I tried creating a new deck and moving there all my cards, and the issue still holds. Then I tried yet another new deck, this time with "10 1", and it didn't work. I was wondering if the issue could derive from one step being lower the the previous one, but since "10 5" sometimes work it doesn't seem to be that.

This is something that also affectes Anki Desktop; if something else come to mind I'll update here, but I really have no clue about what's happening and I have no idea about how to further investigate this.

**Comment by Mornon** [comment]
(I deleted the previous messages and did a summary, sorry for the multiple posts I did.)

I tried some debug, and I found is that if I reset a card and that card is in a subdeck, then every new card in that subdeck will have issue with time steps; still it's not all, since it's not consistent: I did a new deck, moved there all my cards, created a new card, and it had the same issue.

What I tried: I tried to make a new option group with the same setting, just in case something was corrupted, but it didn't work; same goes with creating new decks and moving there all my cards.

I tried to change the intervals for new cards, to no avail; also using the intervals I have on another deck which doesn't present this issue didn't work. Then I tried using "10 5", but I gave up after 6 hours (instead of 5 minutes).

I tried to create the same card in a new deck in the same option group and it worked, then I created in again in the deck with the issue, to no avail: 5 minutes expected, after 10 minutes nothing.

So it seems new decks aren't affected, yet again I tried creating a new deck and moving there all my cards, and the issue still holds. Then I tried yet another new deck, this time with "10 1", and it didn't work. I was wondering if the issue could derive from one step being lower the the previous one, but since "10 5" sometimes work it doesn't seem to be that.

This is something that also affectes Anki Desktop; if something else come to mind I'll update here, but I really have no clue about what's happening and I have no idea about how to further investigate this.
